homeassistant/components/tplink/switch.py

```diff
@@ -29,7 +30,7 @@ async def async_setup_entry(
     device = coordinator.device
     if not device.is_plug and not device.is_strip:
         return
-    entities = []
+    entities: list = []
     if device.is_strip:
         # Historically we only add the children if the device is a strip
         _LOGGER.debug("Initializing strip with %s sockets", len(device.children))
```

Comment:
I checked `python-kasa` and there does not seem to be individual socket support for it, if I'm understanding this correctly: https://github.com/python-kasa/python-kasa/blob/master/kasa/smartstrip.py#L281-L286

This makes sense in a way, since the "LED" element is the status of the plug/strip itself, but not of individual sockets (which could be turned off individually, but I don't believe they have a status LED each ?)

Reply:
At least KP303 has individual leds besides the main one. But you're right, the state information is not being exposed so it makes no sense to expose these controls to homeassistant for the time being. I created an upstream issue for this: https://github.com/python-kasa/python-kasa/issues/250

Reply:
And they actuall do have a status LED each 🤷‍♂️ my bad hehe

Reply:
^ I wrote this before seeing your response, I actually went looking at the products to clarify what was what!
I believe once the lib is updated to handle things properly for the individual plugs, that'll be a one-liner update to this to handle individual LED control!

(I initially was thinking of light strips instead of power strips, hence my confusion 🤦‍♂️)

Reply:
Yeah.. Anyway, let's mark this as resolved as there is currently no way to get that state information via python-kasa and I'm not sure how much use separate controls would be anyway.